src/Microsoft.AspNet.Razor.Runtime/TagHelperContent.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    /// <summary>
+    /// Abstract class which is used to buffer content returned by <see cref="TagHelpers.ITagHelper"/>.
+    /// </summary>
+    public abstract class TagHelperContent
+    {
+        /// <summary>
+        /// Indicates whether the content was modifed.
+        /// </summary>
+        public abstract bool IsModified { get; }
+
+        /// <summary>
+        /// Returns <c>true</c> if <see cref="TagHelperContent"/> is empty, <c>false</c> otherwise.
+        /// </summary>
+        public abstract bool IsEmpty { get; }
+
+        /// <summary>
+        /// Returns <c>true</c> if <see cref="TagHelperContent"/> is whitespace, <c>false</c> otherwise.
+        /// </summary>
+        public abstract bool IsWhiteSpace { get; }
+
+        /// <summary>
+        /// Sets the value of <see cref="TagHelperContent"/>.
+        /// </summary>
+        /// <param name="value">The value to which <see cref="TagHelperContent"/> must be set.</param>
+        public abstract void SetContent(string value);
+
+        /// <summary>
+        /// Appends to the existing value of <see cref="TagHelperContent"/>.
+        /// </summary>
+        /// <param name="value">The value which must be appended.</param>
+        public abstract void Append(string value);
+
+        /// <summary>
+        /// Clears the content.
+        /// </summary>
+        public abstract void Clear();
+
+        /// <summary>
+        /// Gets the content of <see cref="TagHelperContent"/>.
+        /// </summary>
+        /// <returns>A string containing the content of <see cref="TagHelperContent"/>.</returns>
+        public abstract string GetContent();
+
+        /// <summary>
+        /// Copies the contents of the current <see cref="TagHelperContent"/> to another.
+        /// </summary>
+        /// <param name="tagHelperContent">The <see cref="TagHelperContent"/> to which the
+        /// data has to be copied to.</param>
+        public abstract void CopyTo(TagHelperContent tagHelperContent);
```

Comment:
in the MVC reaction PR (aspnet/Mvc#2099), looks like this method works in the wrong direction or at least is inconsistent with most of this API. suggest instead providing `TagHelperOutput.Append(TagHelperOutput output)` and `TagHelperOutput.Set(TagHelperOutput output)` overloads.